src/components/DatePicker/index.ios.js

```diff
@@ -19,18 +20,29 @@ const datepickerPropTypes = {
 class Datepicker extends React.Component {
     constructor(props) {
         super(props);
-
+        const value = DateUtils.getDateAsText(props.value) || DateUtils.getDateAsText(props.defaultValue) || CONST.DATE.MOMENT_FORMAT_STRING;
         this.state = {
             isPickerVisible: false,
             selectedDate: props.value ? moment(props.value).toDate() : new Date(),
+            value,
```

Comment:
Why are we keeping `value` and `selectedDate`? How are they different?

Reply:
`selectedDate`  contains the instantaneous date selected by picker whereas value contains the finally selected date value.In the old scenario, the updates were handled by `onChange` Handlers, whereas now, since we are using `setNativeProps`, it makes sense to store it as a state variable.